### PR TITLE
Separate build targets for the libraries and binary

### PR DESCRIPTION
**Pull request by shelhamer**
In my usage of ALE I only ever need the C library for the Python interface. For convenience I separated the build targets so that it isn't necessary to compile whichever of the C++ library, C library, and ALE binary are unneeded in a particular use case.

p.s. This is my first attempt at a CMake patch so please double-check that its not in error. I'd be happy to correct it if needed.

### CODE REVIEW

**Comment by mgbellemare** [comment]
Thanks for the update! Was there a particular reason, other than cosmetics, for separating the two builds -- given that both still require building all of the source?

File: CMakeLists.txt [hunk 1]
```diff
@@ -6,6 +6,9 @@ set(ALEVERSION "0.5")
 option(USE_SDL "Use SDL" OFF)
 option(USE_RLGLUE "Use RL-Glue" OFF)
 option(BUILD_EXAMPLES "Build Example Agents" ON)
+option(BUILD_CPP_LIB "Build C++ Shared Library" ON)
+option(BUILD_CLI "Build ALE Command Line Interface" ON)
```
Comment (mgbellemare) [inline CMakeLists.txt]: Is this flag used at all?

File: CMakeLists.txt [hunk 2]
```diff
@@ -6,6 +6,9 @@ set(ALEVERSION "0.5")
 option(USE_SDL "Use SDL" OFF)
 option(USE_RLGLUE "Use RL-Glue" OFF)
 option(BUILD_EXAMPLES "Build Example Agents" ON)
+option(BUILD_CPP_LIB "Build C++ Shared Library" ON)
```
Comment (mgbellemare) [inline CMakeLists.txt]: Setting BUILD_CPP_LIB off but leaving BUILD_EXAMPLES on will fail. Can you add some code that errors in this case?

File: CMakeLists.txt (hunk 1 above)
Comment (shelhamer) [inline CMakeLists.txt]: sorry—I accidentally switched flags part-way through the patch. Fixed in latest commit by `BUILD_ALE_BIN` -> `BUILD_CLI`

File: CMakeLists.txt (hunk 2 above)
Comment (shelhamer) [inline CMakeLists.txt]: Thanks for catching that. I've added a rule to force building the C++ lib when the examples are to be built.

**Comment by shelhamer** [comment]
> Was there a particular reason, other than cosmetics, for separating the two builds -- given that both still require building all of the source?

My Python-only use case skips the build of the C++ library and the command line interface/ALE binary. It's not necessarily dramatic, but it does save a little time.

**Pull request merged**